awx/main/tasks.py

```diff
@@ -149,6 +149,81 @@ def handle_setting_changes(self, setting_keys):
             break
 
 
+def _add_remove_queues(app, instance, worker_queues, worker_name):
+    removed_queues = []
+    added_queues = []
+    ig_names = set([ig.name for ig in instance.rampart_groups.all()])
```

Comment:
I think it would be _slightly_ more performant and localized in the DB to do `set(instance.rampart_groups.values_list('name', flat=True))`